README.md

```diff
@@ -28,207 +41,439 @@ Beyond benchmarking, this solution also **provides guidance on:**
 
-cd ansible
-#Test the connection:
+Choose the appropriate playbook based on your needs:
 
-ansible -i inventory.yaml -m ping hammerdb_oracle_client_vms
+```bash
+# NEW: Conditional installation based on benchmark_tool variable
+ansible-playbook -i inventory.yaml main_setup_conditional_benchmark.yml
 
+# Traditional options (install everything)
+ansible-playbook -i inventory.yaml main_setup_complete_benchmark_suite.yml
 
-#Run the playbook to install all required dependencies.
+# Tool-specific setup
 ansible-playbook -i inventory.yaml main_setup_oracle_hammerdb_benchmark.yml
+ansible-playbook -i inventory.yaml main_setup_oracle_swingbench_benchmark.yml
 ```
 
-Individual playbooks can also be executed for granular updates:
+### 5. Individual Component Installation
 
+For granular control, run individual playbooks:
 
-```shell
-# Install Oracle client
+```bash
+# Install Oracle Instant Client
 ansible-playbook -i inventory.yaml playbooks/oracle-client/install_oracle_client.yml
 
-# Configure tnsnames.ora
+# Configure TNS
 ansible-playbook -i inventory.yaml playbooks/configure-tnsnames/configure_tnsnames.yml
 
-# Install HammerDB and copy scripts
+# Install HammerDB
 ansible-playbook -i inventory.yaml playbooks/setup-hammerdb/install_setup_hammer_db.yml
+
+# Install SwingBench
+ansible-playbook -i inventory.yaml playbooks/setup-swingbench/install_setup_swingbench.yml
 ```
 
+---
+
+## Configuration Management
+
+### Benchmark Tool Selection
+
+The `benchmark_tool` variable acts as an enum to control which tools are installed:
 
-## Running the Benchmark
+- **`all`** (default): Install both HammerDB and SwingBench
+- **`hammerdb`**: Install only HammerDB and its dependencies
+- **`swingbench`**: Install only SwingBench and its dependencies
 
-Post-Ansible execution, your target VM should have the following directory structure (default base path: /opt/HammerDB): All the dependencies will be inside location as you configured the path with `benchmark_scripts` folder will have all the custom scripts to run the benchmark.
-`hammerdb-oracle-tns` will have the tnsnames.ora file.
+You can set this variable in multiple ways:
+
+```bash
+# In inventory.yaml
+benchmark_tool: hammerdb
 
-```shell
-[cloud-user@lrangine-vm01 opt]$ tree -d -L 2 /opt/HammerDB
-/opt/HammerDB
-|-- 4.12
-|   |-- agent
-|   |-- bin
-|   |-- config
-|   |-- images
-|   |-- include
-|   |-- lib
-|   |-- modules
-|   |-- scripts
-|   `-- src
-|-- benchmark_scripts
-|   `-- results
-`-- hammerdb-oracle-tns
+# As command-line variable
+ansible-playbook -i inventory.yaml main_setup_conditional_benchmark.yml -e benchmark_tool=swingbench
 
-13 directories
+# For specific host groups
+ansible-playbook -i inventory.yaml main_setup_conditional_benchmark.yml -e benchmark_tool=all
 ```
 
-Go to folder `/opt/HammerDB/benchmark_scripts` on the client VM.
-### Build the schema
+### Variable Structure
+
+Variables are organized into separate files for better maintainability:
```

Comment:
not sure this is still correct, considering the variables moved under single file